test/upgrade/testing_operations_test.go

```diff
@@ -38,15 +35,12 @@ func newConfig(t *testing.T) (upgrade.Configuration, fmt.Stringer) {
 func newExampleZap() (*zap.Logger, fmt.Stringer) {
 	ec := zap.NewDevelopmentEncoderConfig()
 	ec.TimeKey = ""
-	encoder := zapcore.NewConsoleEncoder(ec)
-	buf := &buffer{
-		Buffer: bytes.Buffer{},
-		Mutex:  sync.Mutex{},
-		Syncer: zaptest.Syncer{},
-	}
-	ws := zapcore.NewMultiWriteSyncer(buf, os.Stdout)
-	core := zapcore.NewCore(encoder, ws, zap.DebugLevel)
-	return zap.New(core).WithOptions(), buf
+	buf := upgrade.NewThreadSafeBuffer()
+	core := zapcore.NewCore(
+		zapcore.NewConsoleEncoder(ec),
+		zapcore.NewMultiWriteSyncer(zapcore.AddSync(buf), os.Stdout),
+		zap.DebugLevel)
+	return zap.New(core), buf
```

Comment:
Could this reuse `NewInMemoryLoggerBuffer` and could we make the ThreadSafeBuffer stuff non-exported as a consequence?